### PR TITLE
Convert ExpressionStep to MLv2

### LINKED ISSUE #35982: FE - Convert ExpressionStep to MLv2

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #35982

Depends on #36533 (blocked by #36507)

### Description

One of e2e tests was failing and was skipped out because we need product input for #36574.

### How to verify

Add/edit/remove custom columns in GUI editor.

image

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/notebook/steps/ExpressionStep.unit.spec.tsx [hunk 1]
```diff
@@ -1,125 +1,95 @@
...
-import { createMockEntitiesState } from "__support__/store";
-import { getMetadata } from "metabase/selectors/metadata";
 
-import type { Expression } from "metabase-types/api";
-import { createMockState } from "metabase-types/store/mocks";
-import {
-  createSampleDatabase,
-  ORDERS_ID,
-} from "metabase-types/api/mocks/presets";
+import * as Lib from "metabase-lib";
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/notebook/steps/ExpressionStep.unit.spec.tsx:4]: I recommend to review this file as a new file (a lot has changed).

File: frontend/src/metabase/query_builder/components/notebook/steps/ExpressionStep.tsx [hunk 2]
```diff
@@ -1,48 +1,103 @@
...
+              updateQuery(nextQuery);
+            }
           }}
           reportTimezone={reportTimezone}
         />
       )}
       isLastOpened={isLastOpened}
-      onRemove={({ name }) => updateQuery(query.removeExpression(name))}
+      onRemove={clause => {
+        const nextQuery = Lib.dropStageIfEmpty(
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/notebook/steps/ExpressionStep.tsx:73]: That seems weird - we don't do this in other places. Should it be done by steps?

**Review by ranquild — COMMENTED** [review]
Comment inline

File: frontend/src/metabase/query_builder/components/notebook/steps/ExpressionStep.tsx (hunk 2 above)
Comment (kamilmielnik) [reply frontend/src/metabase/query_builder/components/notebook/steps/ExpressionStep.tsx]: Removed for now: 1444d12324b76f2dea4e58162b8afea21f2a31b2

**Review by ranquild — APPROVED** [review]

**Pull request merged**